app/scripts/contentscript.js

```diff
@@ -82,6 +80,52 @@ function injectScript(content) {
   }
 }
 
+/**
+ * SERVICE WORKER LOGIC
+ */
+
+const WORKER_KEEP_ALIVE_INTERVAL = 1000;
+const WORKER_KEEP_ALIVE_MESSAGE = 'WORKER_KEEP_ALIVE_MESSAGE';
+const TIME_45_MIN_IN_MS = 45 * 60 * 1000;
+
+/**
+ * Don't run the keep worker alive logic for initial JSON RPC methods.
+ * We should not run keep-alive logic for non-dapp pages.
+ */
+const IGNORE_INIT_METHODS_FOR_KEEP_ALIVE = [
+  MESSAGE_TYPE.GET_PROVIDER_STATE,
+  MESSAGE_TYPE.SEND_METADATA,
+];
+
+let keepAliveInterval;
+let keepAliveTimer;
+
+/**
+ * Running this method will ensure the service worker is kept alive for 45 minutes
+ * First message is sent immediately, subsequent messages are sent at interval of WORKER_KEEP_ALIVE_INTERVAL
+ */
+const runWorkerKeepAliveInterval = () => {
+  if (keepAliveTimer) {
+    clearTimeout(keepAliveTimer);
+  }
+
+  keepAliveTimer = setTimeout(() => {
+    clearInterval(keepAliveInterval);
+  }, TIME_45_MIN_IN_MS);
+
+  if (keepAliveInterval) {
+    clearInterval(keepAliveInterval);
+  }
+
+  browser.runtime.sendMessage({ name: WORKER_KEEP_ALIVE_MESSAGE });
```

Comment:
above send message is required only if keepAliveInterval was not already running in other words extension was dead

Reply:
The thing is, I added this in so that we could send a message immediately, as the one in setInterval will run after the interval has passed.

I'm still a fan of keeping it in for the purpose I just mentioned, but I can remove it if you think it would be better. I could also add a comment elaborating on this line. What do you think?  @jpuri

Reply:
lets leave it there 👍